Foundation/NSUserDefaults.swift

```diff
@@ -7,46 +7,286 @@
+        switch CFGetTypeID(anObj) {
+        case CFStringGetTypeID():
+            return (anObj as! CFStringRef)._nsObject
+            
+        case CFNumberGetTypeID():
+            return (anObj as! CFNumberRef)._nsObject
+            
+        case CFURLGetTypeID():
+            return (anObj as! CFURLRef)._nsObject
+            
+        case CFArrayGetTypeID():
+            return (anObj as! CFArrayRef)._nsObject
+            
+        case CFDictionaryGetTypeID():
+            return (anObj as! CFDictionaryRef)._nsObject
+
+        case CFDataGetTypeID():
+            return (anObj as! CFDataRef)._nsObject
+            
+        default:
+            return getFromRegistered()
+        }
+    }
+    public func setObject(value: AnyObject?, forKey defaultName: String) {
+        guard let value = value else {
+            CFPreferencesSetAppValue(defaultName._cfObject, nil, suite?._cfObject ?? kCFPreferencesCurrentApplication)
+            return
+        }
+        
+        var cfType: CFTypeRef? = nil
+		
+		//FIXME: is this needed? Am I overcomplicating things?
+        //Foundation types
+        if let bType = value as? NSNumber {
+            cfType = bType._cfObject
+        } else if let bType = value as? NSString {
+            cfType = bType._cfObject
+        } else if let bType = value as? NSArray {
+            cfType = bType._cfObject
+        } else if let bType = value as? NSDictionary {
+            cfType = bType._cfObject
+        } else if let bType = value as? NSURL {
+			setURL(bType, forKey: defaultName)
+			return
+        } else if let bType = value as? NSData {
+            cfType = bType._cfObject
+            //Swift types
+        } else if let bType = value as? String {
+            cfType = bType._cfObject
+        } else if let bType = value as? Int {
+            cfType = NSNumber(integer: bType)._cfObject
+        } else if let bType = value as? UInt {
+            cfType = NSNumber(unsignedInteger: bType)._cfObject
+        } else if let bType = value as? Int32 {
+            cfType = NSNumber(int: bType)._cfObject
+        } else if let bType = value as? UInt32 {
+            cfType = NSNumber(unsignedInt: bType)._cfObject
+        } else if let bType = value as? Int64 {
+            cfType = NSNumber(longLong: bType)._cfObject
+        } else if let bType = value as? UInt64 {
+            cfType = NSNumber(unsignedLongLong: bType)._cfObject
+        } else if let bType = value as? Bool {
+            cfType = NSNumber(bool: bType)._cfObject
+        } else if let bType = value as? [NSObject: AnyObject] {
+            cfType = bType._cfObject
+        } else if let bType = value as? [String: AnyObject] {
+            cfType = bType.map({ (str, obj) -> (NSString, AnyObject) in
+                return (str._nsObject, obj)
+            })._cfObject
+        } else if let bType = value as? [AnyObject] {
+            cfType = bType._cfObject
+        } else if let bType = value as? [String] {
```

Comment:
if value is an `AnyObject` it cannot be `[String]` or `[AnyObject]` it can only be NSArray etc